app/controllers/idv/otp_delivery_method_controller.rb

```diff
@@ -13,11 +13,8 @@ def new; end
     def create
       result = @otp_delivery_selection_form.submit(otp_delivery_selection_params)
```

Comment:
Thanks for starting this separation. One thing I noticed that I hadn't before is that we probably shouldn't be using the `OtpDeliverySelectionForm` here. Most of the logic only applies to 2FA in the authentication context. For example, we don't want to risk changing the user's OTP delivery preference, and the check for `unsupported_phone?` is redundant because the phone was already validated in the previous step in `IdV::PhoneForm`.

Reply:
I can build out a IdV specific form here, though I think we may want to open another PR for that to keep this one small.

Reply:
Did this here: https://github.com/18F/identity-idp/pull/2440/files